src/loaders/genome_collection/parse_computed_genome_attribs.py

```diff
@@ -133,16 +192,366 @@ def _read_tool_result(result_dir, batch_dir, kbase_collection, load_ver, tool_fi
+        trait_description: str,
+        trait_category: str,
+        trait_type: str):
+    # Append a new trait to the global traits dictionary or check if the trait information is consistent
+
+    if trait_id not in global_traits_meta:
+        # add new trait
+        global_traits_meta[trait_id] = {
+            _SYS_TRAIT_INDEX: trait_index,
+            _SYS_TRAIT_NAME: trait_name,
+            _SYS_TRAIT_DESCRIPTION: trait_description,
+            _SYS_TRAIT_CATEGORY: trait_category,
+            _SYS_TRAIT_TYPE: trait_type,
+        }
+    else:
+        # check if the trait information is consistent
+        existing_trait = global_traits_meta[trait_id]
+        if (existing_trait[_SYS_TRAIT_NAME] != trait_name or
+                existing_trait[_SYS_TRAIT_INDEX] != trait_index or
+                existing_trait[_SYS_TRAIT_DESCRIPTION] != trait_description or
+                existing_trait[_SYS_TRAIT_CATEGORY] != trait_category or
+                existing_trait[_SYS_TRAIT_TYPE] != trait_type):
+            raise ValueError(f'Inconsistent trait information for trait {trait_id}')
+
+
+def _parse_categories(traits_meta: dict[str, dict[str, str]]) -> list[ColumnCategory]:
+    # Parse the trait categories from the global traits dictionary
+
+    categories = {}
+    # loop over each trait in traits_meta
+    for trait in traits_meta.values():
+        trait_category = trait[_SYS_TRAIT_CATEGORY]
+
+        if trait[_SYS_TRAIT_TYPE] == 'count':
+            trait_type = ColumnType.COUNT.value
+        elif trait[_SYS_TRAIT_TYPE] == 'binary':
+            trait_type = ColumnType.BOOL.value
+        else:
+            raise ValueError(f'Unknown trait type {trait[_SYS_TRAIT_TYPE]}')
+
+        # if the trait_category is not available, create a new ColumnCategory object with empty columns
+        if trait_category not in categories:
+            categories[trait_category] = ColumnCategory(category=trait_category, columns=[])
+
+        # add the trait to the appropriate category's list of columns
+        categories[trait_category].columns.append(ColumnInformation(id=str(trait[_SYS_TRAIT_INDEX]),
+                                                                    name=trait[_SYS_TRAIT_NAME],
+                                                                    description=trait[_SYS_TRAIT_DESCRIPTION],
+                                                                    type=trait_type))
+
+    # sort columns in each ColumnCategory object by the column id
+    for category in categories.values():
+        category.columns = sorted(category.columns, key=lambda column: int(column.id))
+
+    # sort ColumnCategory objects by the column id of the first column in each ColumnCategory object
+    sorted_categories = sorted(categories.values(), key=lambda category: int(category.columns[0].id))
+
+    column_ids = [column.id for category in sorted_categories for column in category.columns]
+    if not _ensure_list_ordered(column_ids):
```

Comment:
Maybe add a note here saying that this step might have to be skipped for other trait products in the future, but it works for microtrait

Reply:
👍

Reply:
:+1: